pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,92 @@
+package backuphelpers
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	prune "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+)
+
+type Enqueueable interface {
+	Enqueue()
+}
+
+type BackupVar interface {
+	AddListener(listener Enqueueable)
+	SetBackupSpec(spec *backupv1alpha1.EtcdBackupSpec)
+	ArgString() string
+}
+
+type BackupConfig struct {
+	enabled   bool
+	spec      *backupv1alpha1.EtcdBackupSpec
+	listeners []Enqueueable
+	mux       sync.Mutex
+}
+
+func NewDisabledBackupConfig() *BackupConfig {
+	return &BackupConfig{
+		enabled: false,
+		mux:     sync.Mutex{},
+	}
+}
+
+func (b *BackupConfig) SetBackupSpec(spec *backupv1alpha1.EtcdBackupSpec) {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	if reflect.DeepEqual(b.spec, spec) {
+		return
+	}
+
+	b.enabled = spec != nil
+
+	b.spec = spec
```

Comment:
don't we have to make a copy here?

Reply:
so I make a copy in the periodicController, and pass a pointer to the copy here

Reply:
this way 

```
if item.Name == "default" {
			specClone := item
			c.backupVarGetter.SetBackupSpec(&specClone.Spec.EtcdBackupSpec)
		}
```

Reply:
yes, but should every caller of SetBackupSpec know that they need to make a copy before?

Reply:
ok, I am trying to avoid passing a ref around 

I make the copy within BackupVar then 👍🏽

Reply:
Thank you.

Reply:
done